### PR TITLE
fix: broken missing reactions modal

### PR DESCRIPTION
**Pull request by inghylt**
This PR closes issue #648

Please test out MapViewer and see if everything is working as it should. Please try out edge cases, such as switching between 2d and 3d, selecting another map while inside 2d and 3d and so on, to see if there are any sneaky bugs I have missed.

The modal (now in a separate component called MissingReactionModal) was moved to not be a child under `<div id="mapSidebar__header"...>` to avoid the bug introduced by the new styling in commit 8f43b0df6104b255bd9f9a60b9be8b46c683bb33. Some related information is available at https://stackoverflow.com/questions/31646746/z-index-behaviour-is-different-in-chrome-to-firefox/31710969#31710969

### CODE REVIEW

File: frontend/src/components/explorer/mapViewer/SidebarDataPanels.vue [hunk 1]
```diff
@@ -12,69 +12,12 @@
...
-                        ... and {{ mapReactionList.length - displayedReaction }} more
-                      </button>
-                    </div>
-                  </td>
-                </tr>
-              </tbody>
-            </table>
-          </div>
-          <button class="modal-close is-large" @click="closeModal"></button>
+          <a @click="$emit('update:showModal', true)"> See comparison </a>
```
Comment (e0) [inline frontend/src/components/explorer/mapViewer/SidebarDataPanels.vue:20]: Minor thing and I don't think it was exactly introduced in this PR but we can avoid a bit more repetition with the following.

```suggestion
          <template v-if="currentMap.mapReactionIdSet.length == 1">
            Please note that {{ missingReactionList.length }}
            of the reactions in the {{ currentMap.type }} are not shown on the map.
          </template>
          <template v-else>
            Please note that {{ missingReactionList.length }} of the reactions in the
            {{ currentMap.type }} are not shown on any of the {{ currentMap.name }} maps.
          </template>
          <a @click="$emit('update:showModal', true)"> See comparison </a>
```

**Review by e0 — APPROVED** [review]
Looks great @inghylt! Really clean refactoring. I left a minor suggestion but it's not needed so I'm approving it for now. Have tested in Firefox, Chrome, and Safari on macOS.

File: frontend/src/components/explorer/mapViewer/SidebarDataPanels.vue (hunk 1 above)
Comment (inghylt) [reply frontend/src/components/explorer/mapViewer/SidebarDataPanels.vue]: Nice suggestion! Fixed with 95d825942bddc73f0a804e0751727527fa0e85db

**Review by mihai-sysbio — APPROVED** [review]
Excellent refactoring!
